invenio_app_ils/circulation/tasks.py

```diff
@@ -0,0 +1,32 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2020 CERN.
+#
+# invenio-app-ils is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""Circulation tasks."""
+
+from celery import shared_task
+from celery.utils.log import get_task_logger
+from invenio_circulation.api import Loan
+from invenio_db import db
+
+from invenio_app_ils.circulation.search import get_all_expired_loans
+from invenio_app_ils.proxies import current_app_ils
+
+celery_logger = get_task_logger(__name__)
+
+
+@shared_task
+def cancel_expired_loan_request():
+    """Cancel loan request after expiration date has passed."""
+    expired_loans = get_all_expired_loans().execute()
+    for hit in expired_loans.hits:
+        loan = Loan.get_record_by_pid(hit.pid)
+        if loan["state"] != "CANCELLED":
+            loan["state"] = "CANCELLED"
```

Comment:
To cancel a loan, you need to trigger a transition and not modify the loan directly. For example:
```
params = dict(
    cancel_reason=...
)
current_circulation.circulation.trigger(
            loan, **dict(params, trigger="cancel")
        )
```